Source/DafnyCore/Pipeline/Compilation.cs

```diff
@@ -534,8 +534,19 @@ CompilationInput input
 
     if (boogieProofObligationDesc is ProofObligationDescription.ProofObligationDescription dafnyProofObligationDesc) {
```

Comment:
What triggered this change?

Reply:
This type of proof obligation, as well as several others recently added, can only be described by a Dafny expression with some extra explanatory text. This provides a mechanism for that.